pkg/client/node.go

```diff
@@ -336,7 +336,13 @@ func patchServerSpec(node *k3d.Node) error {
 	node.Labels[k3d.LabelServerAPIHost] = node.ServerOpts.KubeAPI.Host
 	node.Labels[k3d.LabelServerAPIPort] = node.ServerOpts.KubeAPI.Binding.HostPort
 
-	node.Args = append(node.Args, "--tls-san", node.ServerOpts.KubeAPI.Host) // add TLS SAN for non default host name
+	// If the runtime is docker, attempt to use the docker host
+	if runtime.ID() == "docker" && runtime.GetHost() != "" {
+		node.Labels[k3d.LabelServerAPIHostIP] = runtime.GetHost()
+		node.Labels[k3d.LabelServerAPIHost] = runtime.GetHost()
+	}
```

Comment:
The function called here (`GetHost()`) may return an empty string in case of failure (not only, if the value is actually empty), so maybe we should only call it once to not get different values?
That's also overly critical though, so feel free to ignore it.

```suggestion
	if runtime.ID() == "docker" {
	    dockerHost := runtime.GetHost()
	    if dockerHost != "" {
		  node.Labels[k3d.LabelServerAPIHostIP] = dockerHost
		  node.Labels[k3d.LabelServerAPIHost] = dockerHost
		}
	}
```